### PR TITLE
Fusetools2 1193 adapt to new Camel K API related to resource config

### PR DESCRIPTION
**Pull request by apupier**
based on https://github.com/camel-tooling/vscode-camelk/pull/831 merged.

requires Camel 1.5.0 release and updated as default kamel used. merged

Minimal changes to use new API.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ All notable changes to the "vscode-camelk" extension will be documented in this
 - Adapt Didact tutorial registration to VS Code Didact 0.4.0 updated API
 - Update default runtime version to v1.5.0
 - `resource` has been replaced by `resources` to allow providing several of them in Camel K VS Code Tasks
+- Actions related to starting an integration with resource, configmap or secret now require kamel 1.5.0 community or 1.4.0 Red Hat.
```
Comment (bfitzpat) [inline CHANGELOG.md:10]: Couple of minor suggestions:
* missing a final comma in this comma-delimited list (https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/)
* and remove the period at the end to match the other entries in this change log

With these minor adjustments, the line becomes:
"Actions related to starting an integration with resource, configmap, or secret now require kamel 1.5.0 community or 1.4.0 Red Hat"

File: README.md [hunk 2]
```diff
@@ -90,13 +90,13 @@ You can start a new Camel K integration with or without additional options such
 
     The "Basic" option starts the file as a new integration to deploy the integration to the running Kubernetes system.
 
-* **ConfigMap - Apache Camel K Integration with Kubernetes ConfigMap**
+* **ConfigMap - Apache Camel K Integration with Kubernetes ConfigMap as runtime configuration**
```
Comment (bfitzpat) [inline README.md:93]: Would adjust capitalization of the new text to match the previous text: "as Runtime Configuration"

File: README.md [hunk 3]
```diff
@@ -90,13 +90,13 @@ You can start a new Camel K integration with or without additional options such
...
 
     Select a ConfigMap from a list of the published ConfigMaps in your current Kubernetes system. For example:
 
     ![Start Apache Camel K Integration - ConfigMap list](images/kubernetes-configmap-list.jpg)
 
-* **Secret - Apache Camel K Integration with Kubernetes Secret**
+* **Secret - Apache Camel K Integration with Kubernetes Secret as runtime configuration**
```
Comment (bfitzpat) [inline README.md:99]: Would adjust capitalization of the new text to match the previous text: "as Runtime Configuration"

File: package.json [hunk 4]
```diff
@@ -239,7 +239,7 @@
 					},
 					"configmap": {
 						"type": "string",
-						"description": "Configmap key which needs to be already deployed."
+						"description": "Configmap key. The configmap is added as runtime configuration."
```
Comment (bfitzpat) [inline package.json:242]: I find this description a little confusing. The field is the "key" but the configmap is added?

File: package.json [hunk 5]
```diff
@@ -274,6 +274,10 @@
 						"description": "List of additional resources.",
 						"uniqueItems": true
 					},
+					"secret": {
+						"type": "string",
+						"description": "Secret key. The secret is added as runtime configuration."
```
Comment (bfitzpat) [inline package.json:279]: I find this description a little confusing. The field is the "key" but the secret is added?

File: src/IntegrationUtils.ts [hunk 6]
```diff
@@ -32,8 +32,8 @@ import { TelemetryEvent } from '@redhat-developer/vscode-redhat-telemetry/lib';
 const validNameRegex = /^[A-Za-z][A-Za-z0-9\-\.]*(?:[A-Za-z0-9]$){1}/;
 const devModeIntegration: string = 'Dev Mode - Apache Camel K Integration in Dev Mode';
 export const basicIntegration: string = 'Basic - Apache Camel K Integration without extra options';
-export const configMapIntegration: string = 'ConfigMap - Apache Camel K Integration with Kubernetes ConfigMap';
-export const secretIntegration: string = 'Secret - Apache Camel K Integration with Kubernetes Secret';
+export const configMapIntegration: string = 'ConfigMap - Apache Camel K Integration with Kubernetes ConfigMap as runtime configuration';
```
Comment (bfitzpat) [inline src/IntegrationUtils.ts:35]: Would adjust capitalization of the new text to match the previous text: "as Runtime Configuration"

File: src/IntegrationUtils.ts [hunk 7]
```diff
@@ -32,8 +32,8 @@ import { TelemetryEvent } from '@redhat-developer/vscode-redhat-telemetry/lib';
...
+export const secretIntegration: string = 'Secret - Apache Camel K Integration with Kubernetes Secret as runtime configuration';
```
Comment (bfitzpat) [inline src/IntegrationUtils.ts:36]: Would adjust capitalization of the new text to match the previous text: "as Runtime Configuration"

**Review by bfitzpat — COMMENTED** [review]
Some minor questions and suggestions for language adjustments

**Comment by bfitzpat** [comment]
Plus the name of this PR is a bit confusing -- is it "adapt to a new api"?

File: package.json (hunk 4 above)
Comment (apupier) [reply package.json:242]: yes. the configmap corresponding to the key is added to the integration as runtime configuration.

Comment (bfitzpat) [reply package.json:242]: Might be good to spell that out then. It's not clear from the description the relationship between the key and the configmap

**Pull request merged**